### PR TITLE
More Ja Rule improvements

### PR DESCRIPTION
**Pull request by nomis52**

### CODE REVIEW

File: include/ola/rdm/QueueingRDMController.h [hunk 1]
```diff
@@ -84,7 +84,7 @@ class QueueingRDMController: public RDMControllerInterface {
  * The DiscoverableQueueingRDMController also handles discovery, and ensures
  * that only a single discovery or RDM request sequence occurs at once.
  *
- * In this model discovery has a higher precedence that RDM messages.
+ * In this model discovery has a higher precedence than RDM messages.
```
Comment (peternewman) [inline include/ola/rdm/QueueingRDMController.h]: SPaG model,

File: tools/ja-rule/ja-rule-controller.cpp [hunk 2]
```diff
@@ -102,7 +121,6 @@ class ControllerImpl : public ola::rdm::DiscoveryTargetInterface,
     RDMCommandSerializer::Pack(*request, data, &rdm_length);
     m_device->SendMessage(OpenLightingDevice::RDM_REQUEST, data, rdm_length);
 
-    // TODO(simon): do we need to check there isn't already one in-flight?
```
Comment (peternewman) [inline tools/ja-rule/ja-rule-controller.cpp]: Where is this being done?

File: tools/ja-rule/ja-rule-controller.cpp [hunk 3]
```diff
@@ -274,12 +298,11 @@ class Controller : public ola::rdm::DiscoverableRDMControllerInterface {
...
                          bool ok, const UIDSet& uids) {
     OLA_DEBUG << "Discovery complete: " << uids;
-    if (ok) {
-      m_uids = uids;
-    }
+    m_uids = uids;
     if (callback) {
       callback->Run(m_uids);
     }
+    (void) ok;
```
Comment (peternewman) [inline tools/ja-rule/ja-rule-controller.cpp]: OLA_UNUSED.

File: tools/ja-rule/ja-rule-controller.cpp (hunk 2 above)
Comment (nomis52) [inline tools/ja-rule/ja-rule-controller.cpp]: The next step is to wrap this in a QueuingRDMController, which enforces serialization and also handles ACK overflows.

**Comment by peternewman** [comment]
How do you select a UID if you've got more than 26 nomis52? How about also allowing 0-9, or having some more paging option?

File: tools/ja-rule/ja-rule-controller.cpp (hunk 2 above)
Comment (peternewman) [inline tools/ja-rule/ja-rule-controller.cpp]: Can we leave the TODO in until that's done then?

File: include/ola/rdm/QueueingRDMController.h (hunk 1 above)
Comment (nomis52) [inline include/ola/rdm/QueueingRDMController.h]: Fixed in the next PR

File: tools/ja-rule/ja-rule-controller.cpp (hunk 3 above)
Comment (nomis52) [inline tools/ja-rule/ja-rule-controller.cpp]: Added

**Pull request merged**